EnvDTE.Client/Impl/PropertyImpl/ScalarPropertyImplementation.cs

```diff
@@ -0,0 +1,39 @@
+using System;
+using EnvDTE;
+using JetBrains.Annotations;
+
+namespace JetBrains.EnvDTE.Client.Impl.PropertyImpl;
+
+public abstract class ScalarPropertyImplementation(
+    [NotNull] DteImplementation dte,
+    [NotNull] PropertiesImplementation parent,
+    [NotNull] string key)
+    : Property
+{
+    protected readonly DteImplementation DteImplementation = dte;
+
+    public object Application => null;
+    public DTE DTE => DteImplementation;
+    public Properties Parent => parent;
+    public Properties Collection => parent;
+    public string Name => key;
+
+    public object Object
+    {
+        get => this;
+        set => throw new NotSupportedException();
+    }
```

Comment:
From my reading of the docs on `Property` interface, I believe this is supposed to return `Value` and set the `Value`, and not `this`.